Shell/Options.cpp

```diff
@@ -1738,6 +1744,9 @@ void Options::init()
     _maximumXXNarrows.addProblemConstraint(hasHigherOrder());    
     _maximumXXNarrows.tag(OptionTag::HIGHER_ORDER);
 
+    // TODO we have two ways of enabling function extensionality abstraction atm:
```

Comment:
A TODO? "We should sort this out before merging into master." ;)

Reply:
Yesss absolutely. I'm not quite sure what to do about it:
The old way was the option `-fe`, and now it is its own `-uwa` option, which is better I think. So we could get rid of `-fe abstraction`, but some of the schedules might rely on the old option, right? What do you think?

Reply:
(@MichaelRawson @ibnyusuf Have a look at this comment as well for context)

Reply:
Please see my comment below with regards to breaking higher-order support on current master.

Reply:
Let me try to understand better: is this TODO only about redundancy (having two ways of achieving the same) where the one way is new and the other is a now deprecated-ish HO code (superseded by the ahmed-new-hol branch)?

Reply:
Yess exactly. So the function extensionality abstraction used to be done in a different way before. it was a special mechanism using `VSpecialVar`s and such, but now instead it's being done within the UWA framework. This is why before we had a separate option to turn it on, while now it is just a UWA option variant.  In order to keep compability with old schedules i kept the `-fe abstraction` option.